### PR TITLE
Added automatic os detection on startup

### LINKED ISSUE #58: automating os detection
As a user, I want as much ease of use as possible.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #58

### CODE REVIEW

File: src/classes/os-utils.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { UserConfigJson, OperatingSystems } from "../interfaces/types";
 import { OsCommands } from "./os-commands";
 import fs from "fs";
+import {windows, linux, macos} from 'platform-detect';
```
Comment (anttileppa) [inline src/classes/os-utils.ts]:
```suggestion
import { windows, linux, macos } from 'platform-detect';
```

File: src/classes/os-utils.ts [hunk 2]
```diff
@@ -57,6 +58,21 @@ export default class OsUtils {
     }
   }
 
+  public static detectOS = () : string => {
```
Comment (anttileppa) [inline src/classes/os-utils.ts:66]: Add tsdocs

File: src/classes/os-utils.ts [hunk 3]
```diff
@@ -57,6 +58,21 @@ export default class OsUtils {
...
+    let detectedOS : string;
+    if(windows){
```
Comment (anttileppa) [inline src/classes/os-utils.ts]: Fix formatting (all ifs)
```suggestion
    if (windows) {
```

File: src/classes/os-utils.ts [hunk 4]
```diff
@@ -57,6 +58,21 @@ export default class OsUtils {
...
+      detectedOS = "WINDOWS";
+    }
+    if(linux){
```
Comment (anttileppa) [inline src/classes/os-utils.ts:72]: Add space between if-blocks
```suggestion

    if(linux){
```

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Review by eetupur — APPROVED** [review]

**Pull request merged**